lsp-pyls.el

```diff
@@ -353,53 +359,77 @@ parameters referenced in config."
-   ("pyls.plugins.pydocstyle.addSelect" lsp-pyls-plugins-pydocstyle-add-select)
-   ("pyls.plugins.pydocstyle.addIgnore" lsp-pyls-plugins-pydocstyle-add-ignore)
-   ("pyls.plugins.pydocstyle.convention" lsp-pyls-plugins-pydocstyle-convention)
-   ("pyls.plugins.pydocstyle.enabled" lsp-pyls-plugins-pydocstyle-enabled t)
-   ("pyls.plugins.pycodestyle.maxLineLength" lsp-pyls-plugins-pycodestyle-max-line-length)
-   ("pyls.plugins.pycodestyle.hangClosing" lsp-pyls-plugins-pycodestyle-hang-closing t)
-   ("pyls.plugins.pycodestyle.ignore" lsp-pyls-plugins-pycodestyle-ignore)
-   ("pyls.plugins.pycodestyle.select" lsp-pyls-plugins-pycodestyle-select)
-   ("pyls.plugins.pycodestyle.filename" lsp-pyls-plugins-pycodestyle-filename)
-   ("pyls.plugins.pycodestyle.exclude" lsp-pyls-plugins-pycodestyle-exclude)
-   ("pyls.plugins.pycodestyle.enabled" lsp-pyls-plugins-pycodestyle-enabled t)
-   ("pyls.plugins.pylint.enabled" lsp-pyls-plugins-pylint-enabled t)
-   ("pyls.plugins.pylint.args" lsp-pyls-plugins-pylint-args)
-   ("pyls.plugins.flake8.enabled" lsp-pyls-plugins-flake8-enabled)
-   ("pyls.plugins.flake8.exclude" lsp-pyls-plugins-flake8-exclude)
-   ("pyls.plugins.flake8.filename" lsp-pyls-plugins-flake8-filename)
-   ("pyls.plugins.flake8.hangClosing" lsp-pyls-plugins-flake8-hang-closing)
-   ("pyls.plugins.flake8.ignore" lsp-pyls-plugins-flake8-ignore)
-   ("pyls.plugins.flake8.maxLineLength" lsp-pyls-plugins-flake8-max-line-length)
-   ("pyls.plugins.flake8.select" lsp-pyls-plugins-flake8-select)
-   ("pyls.plugins.flake8.config" lsp-pyls-plugins-flake8-config)
-   ("pyls.plugins.preload.modules" lsp-pyls-plugins-preload-modules)
-   ("pyls.plugins.preload.enabled" lsp-pyls-plugins-preload-enabled t)
-   ("pyls.plugins.mccabe.threshold" lsp-pyls-plugins-mccabe-threshold)
-   ("pyls.plugins.mccabe.enabled" lsp-pyls-plugins-mccabe-enabled t)
-   ("pyls.plugins.jedi_symbols.all_scopes" lsp-pyls-plugins-jedi-symbols-all-scopes t)
-   ("pyls.plugins.jedi_symbols.enabled" lsp-pyls-plugins-jedi-symbols-enabled t)
-   ("pyls.plugins.jedi_signature_help.enabled" lsp-pyls-plugins-jedi-signature-help-enabled t)
-   ("pyls.plugins.jedi_references.enabled" lsp-pyls-plugins-jedi-references-enabled t)
-   ("pyls.plugins.jedi_hover.enabled" lsp-pyls-plugins-jedi-hover-enabled t)
-   ("pyls.plugins.jedi_definition.follow_builtin_imports" lsp-pyls-plugins-jedi-definition-follow-builtin-imports t)
-   ("pyls.plugins.jedi_definition.follow_imports" lsp-pyls-plugins-jedi-definition-follow-imports t)
-   ("pyls.plugins.jedi_definition.enabled" lsp-pyls-plugins-jedi-definition-enabled t)
-   ("pyls.plugins.jedi_completion.include_params" lsp-pyls-plugins-jedi-completion-include-params t)
-   ("pyls.plugins.jedi_completion.enabled" lsp-pyls-plugins-jedi-completion-enabled t)
-   ("pyls.configurationSources" lsp-pyls-configuration-sources)))
+ (append '(("pyls.rope.ropeFolder" lsp-pyls-rope-rope-folder)
```

Comment:
consider splitting this to two separate `lsp-register-custom-settings` instead of using one with `append`.